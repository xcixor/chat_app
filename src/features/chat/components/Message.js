import React from "react";
import PropTypes from "prop-types";
import { Col } from "react-bootstrap";

export default function Message({ isMe, message }) {
	return (
		<Col
			md={isMe ? { span: 5, offset: 7 } : { span: 5 }}
			style={{
				backgroundColor: isMe ? "#E8E8E8" : "#A020F0",
				color: isMe ? "#606060" : "#fff",
				height: "auto",
			}}
			className="mb-2 p-2"
		>
			<h4>{isMe ? "Me" : message.sender}</h4>
			<p>{message.message}</p>
			<span style={{ fontSize: "0.8rem" }}>
				<i>{message.sentAt}</i>
			</span>
		</Col>
	);
}

Message.propTypes = {
	isMe: PropTypes.bool.isRequired,
	message: PropTypes.object.isRequired,
};
