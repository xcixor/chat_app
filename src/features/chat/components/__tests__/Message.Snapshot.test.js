import React from "react";
import Message from "../Message";
import renderer from "react-test-renderer";

it("Renders correctly", () => {
	const message = {
		sender: "Test User",
		sentAt: "5:43",
	};
	const tree = renderer.create(<Message isMe={true} message />);

	expect(tree).toMatchSnapshot();
});
