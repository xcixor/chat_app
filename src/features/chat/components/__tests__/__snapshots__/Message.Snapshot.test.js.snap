// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="mb-2 p-2 col-md-5 offset-md-7"
  style={
    Object {
      "backgroundColor": "#E8E8E8",
      "color": "#606060",
      "height": "auto",
    }
  }
>
  <h4>
    Me
  </h4>
  <p />
  <span
    style={
      Object {
        "fontSize": "0.8rem",
      }
    }
  >
    <i />
  </span>
</div>
`;
